src/fu-util.c

```diff
@@ -2311,6 +2352,153 @@ fu_util_check_polkit_actions (GError **error)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(FuUtilPrivate, fu_util_private_free)
 #pragma clang diagnostic pop
 
+static gchar *
+fu_util_get_history_checksum (FuUtilPrivate *priv, GError **error)
+{
+	g_autoptr(FwupdDevice) device = NULL;
+	g_autoptr(FwupdRelease) release = NULL;
+	g_autoptr(GPtrArray) devices = NULL;
+
+	devices = fwupd_client_get_history (priv->client, NULL, error);
+	if (devices == NULL)
+		return NULL;
+	device = fu_util_prompt_for_device (priv, devices, error);
+	if (device == NULL)
+		return NULL;
+	release = fu_util_prompt_for_release (priv, fwupd_device_get_releases (device), error);
+	if (release == NULL)
+		return NULL;
+	return fu_util_prompt_for_checksum (priv, fwupd_release_get_checksums (release), error);
```

Comment:
I think it's going to be confusing that multiple checksums are going to be available to block.  Why not just block the largest (and presumably most secure) checksum?

Reply:
Valid point. We can use the already-exported `fwupd_checksum_get_best()` -- Fixed.